prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml

```diff
@@ -15,6 +15,9 @@ presubmits:
             args:
               - --config-path=./prow/config.yaml
               - --job-config-path=./prow/prowjobs
+              - --plugin-config=prow/plugins.yaml
+              - --strict
+              - --exclude-warning=mismatched-tide
```

Comment:
what does this do?

Reply:
This part is a little weird. There are two tide mismatch warnings, the one we are excluding and the lenient one we are still running: 
https://github.com/kubernetes/test-infra/blob/191b281468351f16bc560e55ee17b4a590db433c/prow/cmd/checkconfig/main.go#L88-L89

The `mistmatched-tide-lenient` warning ensures that all repos that configure Tide to require a label that is associated with a plugin also enable that plugin. This catches the case where you require `lgtm` to merge, but forget to enable the `lgtm` plugin.

The `mismatched-tide` warning **also** checks that all repos that configure Tide to forbid a label that is associated with a plugin also enable that plugin. This catches the case where you forbid `do-not-merge/work-in-progress`, but for get to enable the `wip` plugin.

I don't really like this check because we often intentionally make a Tide query cover multiple repos where one of the repos enables a plugin that manages a blocker label, but the other repos don't. Tide behaves properly because there are never any PRs with the forbidden label in the repo where the blocker plugin is not enabled and this pattern is pretty nice for keeping config readable so I don't consider it a misconfiguration and don't want checkconfig to ban it.

Reply:
interesting, thanks!

I wonder if it might be better to use an explicit lint list or adjust the defaults, golang-ci lint has some flags that help with this (`--disable-all`)

might also be worth a shorter comment in the config file

Reply:
We support either a whitelist or blacklist for specifying which warnings to run: https://github.com/kubernetes/test-infra/blob/191b281468351f16bc560e55ee17b4a590db433c/prow/cmd/checkconfig/main.go#L152-L153
An explicit lint list would be less prone to breaking, but I worry that we'd add new important checks as we add new component/functionality and forget to update the job spec. If there are backwards incompatible checks that are added, we can disable them with a 1 line fix.

Either way seems fine to me though. Let me know if you want me to switch to an explicit list.

Reply:
i'm not actually sure what is better, it seems like ideally you wouldn't have disagreement about the defaults though!

Reply:
(and if you do, maybe we should explicitly opt-in to lints)

Reply:
Its just that one warning and I'd like to avoid having to explicitly opting in. 
@stevekuznetsov what do you think about making that check default to disabled? It would be nice to default to running checks that are appropriate for all Prow instances and then opt-in to any additional ones.

Reply:
I like the check :)

Reply:
I do agree that making one of the two default to off would be nice.

Reply:
I understand why some people might want to enable the check, but I think we should make the default behavior of `checkconfig` appropriate for any Prow instance. There are times when it makes sense and should be valid to provide configuration that violates the `mismatched-tide` check. It can be more efficient and easier to read, especially when plugins cannot be configured to have no-op behavior (e.g. [`mergecommitblocker`](https://github.com/kubernetes/test-infra/tree/master/prow/plugins/mergecommitblocker)).
Previous [discussion](https://github.com/kubernetes/test-infra/pull/9395#issuecomment-429483447).